parts/k8s/kubernetesmastervars.t

```diff
@@ -209,11 +211,19 @@
 {{end}}
     "nsgID": "[resourceId('Microsoft.Network/networkSecurityGroups',variables('nsgName'))]",
 {{if AnyAgentUsesVirtualMachineScaleSets}}
+{{if .AgentPoolProfiles}}
     "primaryScaleSetName": "[concat(parameters('orchestratorName'), '-{{ (index .AgentPoolProfiles 0).Name }}-',parameters('nameSuffix'), '-vmss')]",
+{{else}}
+    "primaryScaleSetName": "",
```

Comment:
@awesomenix As mentioned earlier, this is an issue for Windows. @PatrickLang @jsturtevant  Is it feasible to allow for the Windows flows to acknowledge the absence of AgentPoolProfiles and have a MastersOnly setup?

(Note: Apologies for tagging you all on a holiday. This is by no means urgent :) )

Reply:
Hey! Tariq, thank you for taking time to review this PR on holiday, I am looking at reason for windows failures, its because of a flaky test

span>• Failure [1525.715 seconds]‌
Azure Container Cluster using the Kubernetes Orchestrator
/__w/1/s/gopath/src/github.com/Azure/aks-engine/test/e2e/kubernetes/kubernetes_test.go:84‌
  with a windows agent pool
  ‌/__w/1/s/gopath/src/github.com/Azure/aks-engine/test/e2e/kubernetes/kubernetes_test.go:1052‌
    ‌should be able to deploy an iis webserver [It]‌
    ‌/__w/1/s/gopath/src/github.com/Azure/aks-engine/test/e2e/kubernetes/kubernetes_test.go:1053‌

    ‌Expected‌
        <bool>: false
    to be true‌

    /__w/1/s/gopath/src/github.com/Azure/aks-engine/test/e2e/kubernetes/kubernetes_test.go:1077

Its unrelated to this change.

But i completely agree with you this is not a valid scenario, but the script you tagged doesnt even run in during deployment of Linux Master VM. That script runs only during execution of deployment of windows agent VMs in which case we would have specified the necessary variables like primaryScaleSetName and  primaryAvailabilitySetName. So the scenario is something like

1. Generate template for Master Profiles only
2. Deploy
3. Generate template for Agent profiles only
4. Deploy

Reply:
It's very unlikely that all the windows tests would fail with the same error (which seems to be the case right now) if this is a flake.

I am triggering the build again to confirm.

Reply:
Seems like windows e2e never passed looking at the history https://msazure.visualstudio.com/One/_build?definitionId=50661&_a=summary
Can @PatrickLang @jsturtevant look into these failures?

Reply:
Indeed, there is a known issue w/ the IIS tests on all Windows E2E configs right now.

Reply:
Do we know what the issue is? Looks like the pods are starting but external communication is failing. @PatrickLang Is this related to Azure CNI1.17?  

The logs in the point are not super helpful:

```
2019/02/17 05:10:35 #### $ kubectl expose deployment iis-kubernetes-southeastasia-23972-97695 --type LoadBalancer -n default --target-port 80 --port 80 completed in 1.249608524s
STEP‌: Verifying that the service is reachable and returns the default IIS start page‌
2019/02/17 05:18:23 Unable to validate URL http://13.76.182.186 after 3h40m0s, err: &url.Error{Op:"Get", URL:"http://13.76.182.186", Err:(*net.OpError)(0xc0000aa050)}
```